Jenkinsfile

```diff
@@ -57,25 +57,30 @@ node ("ts-engine && heavy && java8") {
         }
     }
 
-    stage('Integration Tests') {
-        try {
-            sh './gradlew --console=plain integrationTest'
-        } finally {
-            junit testResults: '**/build/test-results/integrationTest/*.xml', allowEmptyResults: true
-        }
-    }
-
     stage('Analytics') {
-        sh './gradlew --console=plain check -x test spotbugsmain'
-        recordIssues tool: checkStyle(pattern: '**/build/reports/checkstyle/*.xml')
-        recordIssues tool: spotBugs(pattern: '**/build/reports/spotbugs/main/*.xml', useRankAsPriority: true)
-        recordIssues tool: pmdParser(pattern: '**/build/reports/pmd/*.xml')
-        recordIssues tool: taskScanner(includePattern: '**/*.java,**/*.groovy,**/*.gradle', lowTags: 'WIBNIF', normalTags: 'TODO', highTags: 'ASAP')
+        sh './gradlew --console=plain check -x test'
+        discoverGitReferenceBuild(defaultBranch: 'develop') //TODO: does this also work for PRs with different base branch?
+        recordIssues skipBlames: true, qualityGates: [[threshold: 1, type: 'NEW', unstable: true]], 
+            tool: checkStyle(pattern: '**/build/reports/checkstyle/*.xml')
+        recordIssues skipBlames: true, qualityGates: [[threshold: 1, type: 'NEW', unstable: true]],
+            tool: spotBugs(pattern: '**/build/reports/spotbugs/main/*.xml', useRankAsPriority: true)
+        recordIssues skipBlames: true, qualityGates: [[threshold: 1, type: 'NEW', unstable: true]],
+            tool: pmdParser(pattern: '**/build/reports/pmd/*.xml')
+        recordIssues skipBlames: true, 
+            tool: taskScanner(includePattern: '**/*.java,**/*.groovy,**/*.gradle', lowTags: 'WIBNIF', normalTags: 'TODO', highTags: 'ASAP')
```

Comment:
If you have a bunch of `recordIssues` steps with the same options but different tools, you could refactor to
```yaml
recordIssues skipBlames: true, qualityGates: […], tools: [
  checkStyle(…),
  spotBugs(…),
  …
]
```

Reply:
In that case, we would also use the same quality gates for all tools, right?

Reply:
I think that's how it would work. In which case you'd leave them separate if you wanted to tune the gate thresholds differently for different tools.

I want there to be some way to change the default for a step's option (e.g. `skipBlames`) for all usages of that step in a stage or pipeline, but it doesn't look like there's actually a general way to do that.